wandb/sdk/artifacts/artifact.py

```diff
@@ -2019,16 +2019,23 @@ def _local_path_to_name(self, file_path: str) -> Optional[str]:
     def delete(self, delete_aliases: bool = False) -> None:
         """Delete an artifact and its files.
 
+        If called on a linked artifact (i.e. a member of a portfolio collection): only the link is deleted, and the
+        source artifact is unaffected.
+
         Arguments:
             delete_aliases: If set to `True`, deletes all aliases associated with the artifact.
                 Otherwise, this raises an exception if the artifact has existing
                 aliases.
+                This parameter is ignored if the artifact is linked (i.e. a member of a portfolio collection).
 
         Raises:
             ArtifactNotLoggedError: If the artifact is not logged.
         """
         self._ensure_logged("delete")
-        self._delete(delete_aliases)
+        if self.collection.is_sequence():
+            self._delete(delete_aliases)
+        else:
+            self._unlink()
```

Comment:
lgtm. Curious though - are we not implementing the double check for this method, i.e., to make the user use a 'force' option to carry on with the delete, like we have in the UI?

Reply:
Wait -- to clarify, do you mean on `artifact.unlink()`? (instead of `artifact.delete()`)

If so --
- I misunderstood, my mistake
- yeah I think this is safer to implement here

Reply:
For posterity: discussed offline -- introducing a bool `force` or `confirm` keyword arg (or other confirmation mechanism) is still probably a good idea.  However, it would be a breaking change, so if it does appear in the future, it'd be safer to 
- implement the change outside this pr
- deprecate/communicate the change to minimize disruption to users